goodconf/__init__.py

```diff
@@ -137,60 +179,63 @@ def type_to_str(tp: Type[Any]) -> str:
 
 
 class GoodConf(BaseSettings):
-    def __init__(self, load: bool = False, **kwargs):
+    _config_file: str = PrivateAttr(None)
+
+    def __init__(self, load: bool = False, config_file: str | None = None, **kwargs):
         """
         :param load: load config file on instantiation [default: False].
 
         A docstring defined on the class should be a plain-text description
         used as a header when generating a configuration file.
         """
-        self._config_file = None
+        # At this point __pydantic_private__ is None, so setting self.config_file
+        # raises an error. To avoid this error, explicitly set
+        # __pydantic_private__ to {} prior to setting self._config_file.
+        _object_setattr(self, "__pydantic_private__", {})
+        self._config_file = config_file
+
         # Emulate Pydantic behavior, load immediately
         if kwargs:
             return super().__init__(**kwargs)
         elif load:
             return self.load()
```

Comment:
I wonder if it would make sense to get rid of `self.load()` and always load immediately like pydantic does. We are already backwards incompatible and having `.load()` doesn't get you much imo. At some point it is simply not worth to fight with pydantic :) What do you think @ipmb?